code/game/machinery/computer/robot.dm

```diff
@@ -76,11 +100,30 @@
 		)
 		data["drones"] += list(drone_data)
 
+
+	data["uploads"] = list()
+	for(var/obj/machinery/computer/upload/U as() in GLOB.uploads_list)
+		if(stat & (NOPOWER|BROKEN))
+			continue
+		if(get_virtual_z_level() != U.get_virtual_z_level())
```

Comment:
I think you also need to check if they're instead both on station zlevels (for multiz stations). Look for `ZTRAIT_STATION` or whatever.

Reply:
and done